src/analysis/seminorm.lean

```diff
@@ -533,6 +533,12 @@ nonneg_of_mul_nonneg_left h zero_lt_two
 
 lemma sub_rev : p (x - y) = p (y - x) := by rw [←neg_sub, p.neg]
 
+lemma le_insert : p y ≤ p x + p (x - y) :=
+calc p y = p (x - (x - y)) : by rw sub_sub_cancel
+... ≤ p x + p (x - y) : p.sub_le _ _
+
+lemma le_insert' : p x ≤ p y + p (x - y) := by { rw sub_rev, exact le_insert _ _ _ }
```

Comment:
Why the name `insert` here?

Reply:
see above https://github.com/leanprover-community/mathlib/pull/11791#discussion_r798599914, it's a generalisation of norm_le_insert in analysis.normed.group.basic

I guess it's "insert" because you're forcing to insert x in the path to y

Reply:
Can you add a docstring to that effect? Perhaps
```suggestion
/-- The direct path from 0 to y is shorter than the path with x "inserted" inbetween. -/
lemma le_insert : p y ≤ p x + p (x - y) :=
calc p y = p (x - (x - y)) : by rw sub_sub_cancel
... ≤ p x + p (x - y) : p.sub_le _ _

lemma le_insert' : p x ≤ p y + p (x - y) := by { rw sub_rev, exact le_insert _ _ _ }
```

@PatrickMassot, does that match your memory of the etymology of `norm_le_insert` when you added it in #4099?